packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,467 @@
+      type: 'repository ',
+      rootDir: utils.sourceRootDir,
+    };
+
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.repositoriesDir,
+    );
+    this.artifactInfo.datasourcesDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.datasourcesDir,
+    );
+    this.artifactInfo.modelDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.modelsDir,
+    );
+
+    this.artifactInfo.defaultTemplate = REPOSITORY_CRUD_TEMPLATE;
+
+    this.option('model', {
+      type: String,
+      required: false,
+      description: 'A valid model name',
+    });
+
+    this.option('id', {
+      type: String,
+      required: false,
+      description: 'A valid ID property name for the specified model',
+    });
+
+    this.option('datasource', {
+      type: String,
+      required: false,
+      description: 'A valid datasource name',
+    });
+
+    return super._setupGenerator();
+  }
+
+  setOptions() {
+    return super.setOptions();
+  }
+
+  async checkPaths() {
+    // check for datasources
+    if (!fs.existsSync(this.artifactInfo.datasourcesDir)) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_DIRECTORY} ${this.artifactInfo.datasourcesDir}.
+        ${chalk.yellow(
+          'Please visit https://loopback.io/doc/en/lb4/DataSource-generator.html for information on how datasources are discovered',
+        )}`,
+        ),
+      );
+    }
+
+    // check for models
+    if (!fs.existsSync(this.artifactInfo.modelDir)) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_DIRECTORY} ${this.artifactInfo.modelDir}.
+        ${chalk.yellow(
+          'Please visit https://loopback.io/doc/en/lb4/Model-generator.html for information on how models are discovered',
+        )}`,
+        ),
+      );
+    }
+  }
+
+  async promptDataSourceName() {
+    if (this.shouldExit()) return false;
+
+    debug('Prompting for a datasource ');
+    let cmdDatasourceName;
+    let datasourcesList;
+
+    // grab the datasourcename from the command line
+    cmdDatasourceName = this.options.datasource
+      ? utils.toClassName(this.options.datasource) + 'Datasource'
+      : '';
+
+    debug(`command line datasource is  ${cmdDatasourceName}`);
+
+    try {
+      datasourcesList = await utils.getArtifactList(
+        this.artifactInfo.datasourcesDir,
+        'datasource',
+        true,
+      );
+      debug(
+        `datasourcesList from ${utils.sourceRootDir}/${
+          utils.datasourcesDir
+        } : ${datasourcesList}`,
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    const availableDatasources = datasourcesList.filter(item => {
+      debug(`data source unfiltered list: ${item}`);
+      const result = this._isConnectorOfType(
+        VALID_CONNECTORS_FOR_REPOSITORY,
+        item,
+      );
+      return result;
+    });
+
+    debug(`artifactInfo.dataSourceClass ${this.artifactInfo.dataSourceClass}`);
```

Comment:
This seems out of place. Did you mean to have it after the `if` statement on Line 264?